### ISSUE TITLE
Term's stacktrace is quite (perhaps too) wasteful of terminal real estate.

### ISSUE THREAD

**Issue by emojjon** [FedeClaudi/Term.jl#197]
I am very much in support of Term.jl (and rich.py) in general as I think that all sorts of visual clues makes it easier to find information and frees the brain to concentrate on the issue at hand instead of squinting until you find some information hidden in a haystack of uniform text. So thumbs up in general!

However, I find that the error formatting is a bit lacking. There seems to be very little room for the actual information in the stacktrace, even though nice visual clues show you where it might have been.

![Term_Stacktrace](https://user-images.githubusercontent.com/66027466/214059894-21080624-53d7-44b7-9fcb-39b96e0b1c73.png)
Specifically, with reference to my screen dump:
1. Code lines are elided even though they seem to fit in the panel (the white panel in the picture). At least, more of the line could be shown.
2. This seems to be doubly true of the line that actually contains the error.
3. The white panel is not expanded despite on the one hand having to truncate the code lines and on the other hand having a lot of room inside the blue panel to spare.
4. An estimated 50-60 columns at the end are not used at all (which might be fine, as long as we see what we need).

It could perhaps be said that the blue panel could be wider without making it come closer to the orange panel than what it already is on the left side. There might be other reasons, however, why it is good, practical, aesthetic to do it this way and that's just a few columns anyway.

In short, it would be nice to be able to see more information in the stacktrace and I'm not saying how it's best done but I think there is room (literally) for improvement.

**Comment by FedeClaudi** [comment]
Hi, 
thanks for getting in touch and for the feedback. 
The error display graphics in the upcoming release have actually been re-worked to try to address some of these issues.
This is an example:
<img width="986" alt="image" src="https://user-images.githubusercontent.com/17436313/214078739-c5cb9ed9-8c48-48c1-bb0b-64a64031d393.png">

As you can see the inner "code" panel is now expanded thus removing `3`.  Also the way code is rendered inside the panel is improved leading to less frequently elided code (longer lines should get broken into multiple lines). 
I've also added a link pointing at the source of the problem which is a useful bit of information. 

Another thing worth pointing out is that the stack trace system will also hide frames from code within installed packages. Let's say you have an error in some plotting code, this usually involves about a dozen frames within `Plots`, but 99% of the time the error is in your code, not in an external library, so those frames will be skipped making it easier to find the actual source of the problem. 

Finally, regarding the space to the side, there's an upper limit on the width of the stacktrace of 140 columns. Folks previously mentioned that having too large of a display makes it hard to parse so I've added this in. If you think it would be useful, though, I could add an option to change that.

**Comment by emojjon** [comment]
Veloce come un furetto! Mille grazie!

I think it would be good to be able to allow wider terminals. It's not as if we're getting less screen real estate over time. I can see that people might not want to have it turned on by default if it can cause trouble (no idea what the problem is, but I believe that it's there if people say so). It would be especially good if this option was also documented somehow. Keep up the good work!

**Comment by FedeClaudi** [comment]
From the next release this will work:
```julia
import Term: DEFAULT_STACKTRACE_WIDTH
DEFAULT_STACKTRACE_WIDTH[] = 500
install_term_stacktrace()
1 + "a"
```

so you can set the max width of the stacktrace. It will still fit in the available space if the terminal is narrower than `DEFAULT_STACKTRACE_WIDTH`
